### DISCUSSION TITLE
Reevaluate the uses of `$` and `%`

### DISCUSSION THREAD

**Discussion by f2l2pe** [godotengine/godot-proposals#9051] [Scripting]
Hey all, I'm an experienced programmer learning Godot and gdscript.
In this learning trajectory, I've come in contact with those two operators, `$` and `%`, which are confusing, mainly because I don't get any checks while using them in GdScript.

I had this problem where I was using `$` when I wanted to use `%`, but spotting that is a little bit hard since it's only one character, and they aren't exactly that different.

What if those are removed to keep the code cleaner and easier to read?
When converting some node to an `Access as Unique Name` lets you use the node name as is, without any symbol or have only `%` as maybe using the name as is could cause collisions.

Another option would be having a choice to make using `$` display errors in the editor since this wouldn't break any workflow from other people.

**Comment by f2l2pe** [comment]
A different color would already suffice tbh

![image](https://github.com/godotengine/godot-proposals/assets/6012227/c6710e81-98f5-48e4-abde-00cbac1d0222)

I could open a PR for this

**Comment by theraot** [comment]
I'd be OK with a (opt-in, per project) warning on `$`. I'd probably use it.

**Comment by Mickeon** [comment]
> I had this problem where I was using `$` when I wanted to use `%`, but spotting that is a little bit hard since it's only one character, and they aren't exactly that different.

I would heavily recommend writing `$%`, instead, which is what the symbol is a shorthand of. I have noted a similar concern when this was implemented, but unfortunately for me `%` is here to stay.
